File: BMI v.3.0/app/src/main/java/pl/arisa/bmicalculator/ui/quiz/QuizEndFragment.java
package pl.arisa.bmicalculator.ui.quiz;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import pl.arisa.bmicalculator.MainActivity;
import pl.arisa.bmicalculator.R;

public class QuizEndFragment extends Fragment {

    private static final String ARG_RESULT = "result";
    private static final String ARG_QUESTION_COUNT = "questionsCount";
    private int result;
    private int questionsCount;

    public QuizEndFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            result = getArguments().getInt(ARG_RESULT);
            questionsCount = getArguments().getInt(ARG_QUESTION_COUNT);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_quiz_end, container, false);
        TextView tv_result = view.findViewById(R.id.quiz_result);
        tv_result.setText(getString(R.string.text_result, result, questionsCount, (float) result / questionsCount * 100));

        return view;
    }

}

File: BMI v.4.0/app/src/main/java/pl/arisa/bmicalculator/ui/quiz/QuizEndFragment.kt
package pl.arisa.bmicalculator.ui.quiz

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import pl.arisa.bmicalculator.R

class QuizEndFragment : Fragment() {
    private var result = 0
    private var questionsCount = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        result = requireArguments().getInt(ARG_RESULT)
        questionsCount = requireArguments().getInt(ARG_QUESTION_COUNT)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_quiz_end, container, false)
        val tv_result = view.findViewById<TextView>(R.id.quiz_result)
        tv_result.text = getString(
            R.string.text_result,
            result,
            questionsCount,
            result.toFloat() / questionsCount * 100
        )
        return view
    }

    companion object {
        private const val ARG_RESULT = "result"
        private const val ARG_QUESTION_COUNT = "questionsCount"
    }
}